decentraland/unity-renderer
# fix: avatar names flaky tests & profanity filter into locator service

## What does this PR change?

Fix for avatar name flaky tests. Profanity filter service is now mocked to give an immediate result when trying to filter names. To be able to do so, profanity filter singleton has been migrated into the service locator as a service.

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=fix/names-flaky-tests
2. Open the nearby channel or any other channel and write bad words with the profanity filtering option enabled
3. The words should be filtered accordingly
4. Let a user change their name including a badword
5. The name should be filtered accordingly

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
